qa-tests-backend/scripts/lib.sh

```diff
@@ -2,22 +2,15 @@
 
 # Common functions for deploying a cluster for QA tests
 
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../.. && pwd)"
+
 set -euo pipefail
```

Comment:
How about swapping these lines?

Reply:
> How about swapping these lines?

There is a weird error when the set is before the ROOT determination that I was not able to resolve. This is why this is ordered this way in all the CI YAML -> bash scripts.

Reply:
> There is a weird error when the set is before the ROOT determination that I was not able to resolve

WAT?

```
$ cat a.sh 
#!/bin/bash
set -euo pipefail
ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../.. && pwd)"
echo a
$ ./a.sh 
a
$ echo $?
0
$ 
```

You got me thinking. Can you show an example?

Reply:
Try changing the order in `scripts/ci/lib.sh` and you should see it. I'll see if I can repro/dig up my notes to justify the order.